src/main/java/org/cactoos/iterable/Filtered.java

```diff
@@ -65,7 +65,7 @@ public Filtered(final Func<X, Boolean> fnc, final X... src) {
      * @param fnc Predicate
      * @param src Source iterable
      */
-    public Filtered(final Func<X, Boolean> fnc, final Iterable<X> src) {
+    public Filtered(final Func<X, Boolean> fnc, final Iterable<? extends X> src) {
```

Comment:
@andreoss to be complete, we need `Func<? super X, Boolean>`, so that we can write for example:
```java
 Iterable<Double> res = new Filtered<Double>((Number d) -> d.doubleValue() > 0, new IterableOf<Double>(4.1));
```

That's why I guess we need a few tests..

Reply:
@victornoel Added a test